### PR TITLE
feat: add nightly tiup pipeline related jobs dsl

### PR DESCRIPTION
**Pull request by lijie0123**
## Why:
add job dsl for nightly related jobs, top down from three main pipelines
- release-tidb-master-tiup-nightly
- release-tidb-master-image-nightly
- release-all-images-by-branch-for-internal

and partial release-GA pipelines

## todo list
- upload-pipeline-run-data-to-db: unknown file agent_tibuild_pipeline_run.py
- send_notify: unknown file agent-jenkinsci.py
- tiup_mirror_online_ga: remove password params

### CODE REVIEW

File: jenkins/pipelines/cd/tiup/tiup-mirror-test.groovy [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+    comps.each {
+        tiup_list(it)
+        tiup_install(it, "")
+    }
+    tiup_update_self()
+    comps.each {
+        tiup_update(it, "")
+    }
+}
+try {
```
Comment (wuhuizuo) [inline jenkins/pipelines/cd/tiup/tiup-mirror-test.groovy:69]: please add a blank line

File: jenkins/pipelines/cd/tiup/tiup-check-online-version.groovy [hunk 2]
```diff
@@ -25,8 +40,10 @@ pipeline {
                 }
                 stage('linux/arm64') {
                     agent {
-                        node {
-                            label 'linux && arm64'
+                        kubernetes {
+                            yaml tiupYaml
+                            defaultContainer 'tiup'
+                            nodeSelector "kubernetes.io/arch=arm64"
                         }
```
Comment (wuhuizuo) [inline jenkins/pipelines/cd/tiup/tiup-check-online-version.groovy:47]: does it need set cloud?

Comment (lijie0123) [reply jenkins/pipelines/cd/tiup/tiup-check-online-version.groovy:47]: default clould is ksyun

File: jenkins/pipelines/cd/tiup/tiup-check-online-version.groovy [hunk 3]
```diff
@@ -5,18 +5,33 @@ tiup mirror reset
...
 
+final  tiupYaml='''
+apiVersion: v1
+kind: Pod
+spec:
+  containers:
+  - name: tiup
+    image: hub.pingcap.net/jenkins/tiup
+    args: ["sleep", "infinity"]
+'''
```
Comment (wuhuizuo) [inline jenkins/pipelines/cd/tiup/tiup-check-online-version.groovy:16]: Prefer to use yaml file to manage them.

And we need set tty enabled, no need args.

File: jenkins/pipelines/cd/atom-jobs/upload-pipeline-run-data-to-db.groovy [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+                "pipeline_run_start_time":params.PIPELINE_RUN_START_TIME,
+                "pipeline_run_end_time":params.PIPELINE_RUN_END_TIME,
+        ]
+
+        collect_pipeline_info(finally_result)
+
+    }
+}
+
+
```
Comment (wuhuizuo) [inline jenkins/pipelines/cd/atom-jobs/upload-pipeline-run-data-to-db.groovy:88]: Please check the script style.

File: jenkins/pipelines/cd/atom-jobs/upload-pipeline-run-data-to-db.groovy [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+                "jenkins_run_url": params.JENKINS_RUN_URL,
+                "pipeline_revoker": params.PIPELINE_REVOKER,
+                "error_code": params.ERROR_CODE,
+                "error_summary":params.ERROR_SUMMARY,
+                "pipeline_run_start_time":params.PIPELINE_RUN_START_TIME,
+                "pipeline_run_end_time":params.PIPELINE_RUN_END_TIME,
+        ]
+
+        collect_pipeline_info(finally_result)
+
```
Comment (wuhuizuo) [inline jenkins/pipelines/cd/atom-jobs/upload-pipeline-run-data-to-db.groovy:84]: Please trim the latest blank line.

File: jenkins/pipelines/cd/atom-jobs/upload-pipeline-run-data-to-db.groovy [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+                        name: 'JENKINS_RUN_URL',
+                        description: '',
+                        trim: true
+                ),
+                string(
+                        defaultValue: '',
+                        name: 'PIPELINE_REVOKER',
+                        description: '',
+                        trim: true
+                ),
```
Comment (wuhuizuo) [inline jenkins/pipelines/cd/atom-jobs/upload-pipeline-run-data-to-db.groovy:40]: Please do not define params in pipeline, migrate them to job DSL, and use simple one line style for multi params.

Comment (lijie0123) [reply jenkins/pipelines/cd/atom-jobs/upload-pipeline-run-data-to-db.groovy:40]: one-line style can not set trim

File: jenkins/pipelines/cd/atom-jobs/send_notify.groovy [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        container("golang") {
+            writeJSON file: 'result.json', json: result, pretty: 4
+            sh 'cat result.json'
+            archiveArtifacts artifacts: 'result.json', fingerprint: true
+            sh """
+            export LC_CTYPE="en_US.UTF-8"
+            wget ${FILE_SERVER_URL}/download/rd-atom-agent/agent-jenkinsci.py
+            python3 agent-jenkinsci.py result.json || true
+            """
+            // 只进行失败通知
```
Comment (wuhuizuo) [inline jenkins/pipelines/cd/atom-jobs/send_notify.groovy:40]: comment with english, keep style same.

File: jenkins/pipelines/cd/tiup/tiup-check-online-version.groovy (hunk 3 above)
Comment (lijie0123) [reply jenkins/pipelines/cd/tiup/tiup-check-online-version.groovy:16]: it will fail
and tty will use resource on host

**Review by wuhuizuo — APPROVED** [review]
/lgtm
/approve

**Pull request merged**